cmd/ipfs/daemon.go

```diff
@@ -109,6 +113,14 @@ func daemonFunc(req cmds.Request, res cmds.Response) {
 		}
 	}()
 
+	// check transport encryption flag.
+	unencrypted, _, _ := req.Option(unencryptTransportKwd).Bool()
+	if unencrypted {
+		log.Warningf(`Running with --%s: All connections are UNENCRYPTED.
+		You will not be able to connect to regular encrypted networks.`, unencryptTransportKwd)
+		conn.EncryptConnections = false
```

Comment:
> Exposed package globals like this frighten me. Will we remove this once node interop works?
> If it stays around, I'd prefer conn.DisableEncryption()

I imagine with a function we can also make it one-way-- i.e. can only disable it or something.

btw, nothing prevents people from writing different implementations -- or, if they have access to the process, overwrite random memory. Even without this switch, if you can manipulate the process everything's done.

> I'd still prefer if this was an opt-in for the lifetime of the process. No funky on/off switching.

yeah it is